### PR TITLE
API_Implement_UpdateUserDto

### PR DESCRIPTION
**Pull request by Yuriy-SE**
- Implement_UpdateUserDto
- prettier
- additional password validation

### CODE REVIEW

File: src/user/dto/update-user.dto.ts [hunk 1]
```diff
@@ -1,4 +1,39 @@
...
-export class UpdateUserDto extends PartialType(CreateUserDto) {}
+export class UpdateUserDto extends PartialType(CreateUserDto) {
+  @IsOptional()
+  @ApiProperty({ example: 'user@example.com' })
+  @IsEmail()
+  email: string;
+
+  @ApiProperty({
+    example: 'password123',
+    description:
```
Comment (Elgrueber) [inline src/user/dto/update-user.dto.ts:20]: Passport must be at least 6 characters or more than 6 symbols?

File: src/user/dto/update-user.dto.ts [hunk 2]
```diff
@@ -1,4 +1,39 @@
...
+    message: 'Password must contain at least one lowercase letter!',
+  })
+  @Matches(/[A-Z]/, {
+    message: 'Password must contain at least one uppercase letter!',
+  })
+  @Matches(/[!@#$%^&*()_+\-=\[\]{};:'",.<>?/\\|`~]/, {
+    message: 'Password must contain at least one special character!',
+  })
+  password: string;
+}
```
Comment (Elgrueber) [inline src/user/dto/update-user.dto.ts:39]: We have copy paste in 2 files - is it ok?

File: src/user/dto/update-user.dto.ts (hunk 1 above)
Comment (Yuriy-SE) [reply src/user/dto/update-user.dto.ts:20]: usually 'characters' no 'symbols'. about the length: at least 6 and no more than 20 

`@MinLength(6, { message: 'Password must be more than 6 symbols!' })
  @MaxLength(20, { message: 'Password must be at most 20 characters long!' })`

File: src/user/dto/update-user.dto.ts (hunk 2 above)
Comment (Yuriy-SE) [reply src/user/dto/update-user.dto.ts:39]: it is. because it's same password validation.

Comment (Yuriy-SE) [reply src/user/dto/update-user.dto.ts:39]: first one when we create a user and second one when we update a user

File: src/user/dto/update-user.dto.ts [hunk 3]
```diff
@@ -1,4 +1,39 @@
...
+  @IsEmail()
+  email: string;
+
+  @ApiProperty({
+    example: 'password123',
+    description:
+      'Password must be at least 6 characters long and no more than 20 characters long! Password must contain at least one digit and  at least one lowercase letter and at least one uppercase letter!',
+  })
+  @IsOptional()
+  @MinLength(6, { message: 'Password must be more than 6 symbols!' })
```
Comment (Elgrueber) [inline src/user/dto/update-user.dto.ts]: Perhaps you could update the message to 'Password must be at least 6 characters,' as the current message isn't quite accurate.

Comment (Yuriy-SE) [reply src/user/dto/update-user.dto.ts]: typo ) but i'll update

Comment (Yuriy-SE) [reply src/user/dto/update-user.dto.ts]: updated

File: src/user/dto/update-user.dto.ts (hunk 2 above)
Comment (Elgrueber) [reply src/user/dto/update-user.dto.ts:39]: I understand it. But better to avoid duplication, using objects, for example

**Review by Elgrueber — DISMISSED** [review]

**Review by Elgrueber — APPROVED** [review]

**Pull request merged**